packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -168,6 +168,20 @@ contract OPContractsManager is ISemver {
     /// which is intended to be DELEGATECALLed.
     OPContractsManager internal immutable thisOPCM;
 
+    /// @notice The address of the upgrade controller.
+    address public immutable upgradeController;
+
+    /// @notice The prestate hash to be used for the upgrade.
+    bytes32 public immutable prestateHash;
```

Comment:
Per @ajsutton, I think this should be an input per-chain, just to be safe? https://discord.com/channels/1244729134312198194/1332055053003001886/1332084289537441912